atlasdb-client/src/main/java/com/palantir/atlasdb/util/AtlasDbMetrics.java

```diff
@@ -39,16 +45,38 @@ public static synchronized void setMetricRegistry(MetricRegistry metricRegistry)
 
     // Using this means that all atlasdb clients will report to the same registry, which may give confusing stats
     public static MetricRegistry getMetricRegistry() {
-        if (AtlasDbMetrics.metrics == null) {
+        MetricRegistry currentMetricRegistry = AtlasDbMetrics.metrics;
+        if (currentMetricRegistry == null) {
             synchronized (AtlasDbMetrics.class) {
-                if (AtlasDbMetrics.metrics == null) {
-                    AtlasDbMetrics.metrics = SharedMetricRegistries.getOrCreate(DEFAULT_REGISTRY_NAME);
-                    log.info("Metric Registry was not set, setting to default registry name of "
+                currentMetricRegistry = AtlasDbMetrics.metrics;
+                if (currentMetricRegistry == null) {
+                    currentMetricRegistry = SharedMetricRegistries.getOrCreate(DEFAULT_REGISTRY_NAME);
+                    log.warn("Metric Registry was not set, setting to shared default registry name of "
```

Comment:
The intent with `currentMetricRegistry` is to minimize the happy read path to a single volatile read, where previously we would do at least 2 volatile reads when the registry was already set.